server/schedule/filter/filters.go

```diff
@@ -40,7 +40,9 @@ func SelectSourceStores(stores []*core.StoreInfo, filters []Filter, opt *config.
 				targetID := ""
 				filterCounter.WithLabelValues("filter-source", s.GetAddress(),
 					sourceID, filters[i].Scope(), filters[i].Type(), sourceID, targetID).Inc()
-				collector.Collect(plan.SetResource(s), plan.SetStatus(status))
+				if collector != nil {
+					collector.Collect(plan.SetResource(s), plan.SetStatus(status))
```

Comment:
Do you need to check it? I think already do it in the `Collect` function.

Reply:
Because `plan.SetResource(s)` and `SetResource` will be executed before executing `Collect ` and they will create `newObject` for `Option`